pkg/globalmanager/controllers/jointinference/jointinferenceservice.go

```diff
@@ -422,113 +351,234 @@ func (c *Controller) createWorkers(service *sednav1.JointInferenceService) (acti
+	oldService := old.(*sednav1.JointInferenceService)
+	newService := cur.(*sednav1.JointInferenceService)
+
+	// Check if the cloud worker configuration has changed
+	if !reflect.DeepEqual(oldService.Spec.CloudWorker, newService.Spec.CloudWorker) {
+		// Update cloud worker and log any errors
+		if err := c.updateCloudWorker(newService); err != nil {
+			klog.Errorf("Failed to update cloud worker for service %s/%s: %v", newService.Namespace, newService.Name, err)
+		}
+	}
+
+	// Retrieve the address of the cloud inference service
+	var bigModelHost string
+	svc, err := c.kubeClient.CoreV1().Services(oldService.Namespace).Get(context.Background(),
+		strings.ToLower(oldService.Name+"-"+jointInferenceForCloud), metav1.GetOptions{})
 	if err != nil {
-		return fmt.Errorf("failed to get edge model %s: %w",
-			edgeModelName, err)
+		if errors.IsNotFound(err) {
+			// If the service does not exist, create a new one and retrieve its address
+			klog.Info("Cloud service not found, creating new service...")
+			bigModelHost, err = runtime.CreateEdgeMeshService(c.kubeClient, oldService, jointInferenceForCloud, BigModelPort)
+			if err != nil {
+				klog.Errorf("Failed to create EdgeMesh service for service %s/%s: %v", oldService.Namespace, oldService.Name, err)
+			}
+		} else {
+			klog.Errorf("Failed to get cloud service %s/%s: %v", oldService.Namespace, oldService.Name, err)
+		}
+	} else {
+		bigModelHost = fmt.Sprintf("%s.%s", svc.Name, svc.Namespace)
 	}
 
-	secretName := edgeModel.Spec.CredentialName
-	var modelSecret *v1.Secret
-	if secretName != "" {
-		modelSecret, _ = c.kubeClient.CoreV1().Secrets(service.Namespace).Get(context.TODO(), secretName, metav1.GetOptions{})
+	// Check if the edge worker configuration has changed
+	if !reflect.DeepEqual(oldService.Spec.EdgeWorker, newService.Spec.EdgeWorker) {
+		// Update edge worker and log any errors
+		if err := c.updateEdgeWorker(newService, bigModelHost); err != nil {
+			klog.Errorf("Failed to update edge worker for service %s/%s: %v", newService.Namespace, newService.Name, err)
+		}
 	}
 
-	edgeWorker := service.Spec.EdgeWorker
-	HEMParameterJSON, _ := json.Marshal(edgeWorker.HardExampleMining.Parameters)
-	HEMParameterString := string(HEMParameterJSON)
+	return nil
+}
 
+func (c *Controller) createOrUpdateWorker(service *sednav1.JointInferenceService, workerType string, bigModelHost string, bigModelPort int32, create bool) error {
+	var modelName string
+	var modelTemplate v1.PodTemplateSpec
 	var workerParam runtime.WorkerParam
 
+	// Set the corresponding parameters according to the workerType.
+	switch workerType {
+	case jointInferenceForCloud:
+		modelName = service.Spec.CloudWorker.Model.Name
+		modelTemplate = *service.Spec.CloudWorker.Template.DeepCopy()
+
+		workerParam.Env = map[string]string{
+			"BIG_MODEL_BIND_PORT": strconv.Itoa(int(bigModelPort)),
+		}
+		workerParam.WorkerType = workerType
+		workerParam.HostNetwork = false // The cloud does not need HostNetwork.
```

Comment:
Leave the configuration of hostNetwork to the user. There is no need for mandatory configuration here.